appointment-svc/middlewares/auth-middleware.ts

```diff
@@ -1,21 +1,51 @@
 import jwksRsa from 'jwks-rsa';
 import dotenv from 'dotenv';
+import { NextFunction, Request, Response } from 'express';
+import decryptData from '../utils/decryption';
 const { expressjwt: jwt } = require('express-jwt');
 
 dotenv.config();
 
-const authMiddleware = () => {
-  return jwt({
+export interface IMiddlewareRequest extends Request {
+  decryptedData: any;
+}
+
+function authMiddleware(req: Request, res: Response, next: NextFunction) {
+  const jwtCheck = jwt({
     secret: jwksRsa.expressJwtSecret({
       cache: true,
       rateLimit: true,
       jwksRequestsPerMinute: 5,
-      jwksUri: `${process.env.ISSUER_URL}.well-known/jwks.json`,
+      jwksUri: `${process.env.ISSUER_URL}/.well-known/jwks.json`,
     }),
     audience: 'http://localhost:3001',
     issuer: process.env.ISSUER_URL,
     algorithms: ['RS256'],
   });
-};
+
+  jwtCheck(req, res, (err: any) => {
+    if (err) {
+      const { hash, encryptionIV } = req.body;
```

Comment:
is it because on the frontend you will extract these variables from the URL and send it here in the body of POST request?